libusb.go

```diff
@@ -224,14 +224,21 @@ func (libusbImpl) setDebug(c *libusbContext, lvl int) {
 }
 
 func (libusbImpl) getDeviceDesc(d *libusbDevice) (*DeviceDesc, error) {
-	var desc C.struct_libusb_device_descriptor
+	var desc, pathData, path = C.struct_libusb_device_descriptor{}, [8]uint8{}, []uint8{}
 	if err := fromErrNo(C.libusb_get_device_descriptor((*C.libusb_device)(d), &desc)); err != nil {
 		return nil, err
 	}
+	pathLen := int(C.libusb_get_port_numbers((*C.libusb_device)(d), (*C.uint8_t)(&pathData[0]), 8))
+	if pathLen > 0 {
+		path = pathData[:pathLen-1]
+	}
+	port := int(C.libusb_get_port_number((*C.libusb_device)(d)))
```

Comment:
do I understand correctly that path includes the current port number, as the last element? If so, there's no need to call the C function again, just assign `pathData[pathLen-1]`. Perhaps directly in the struct literal below (line 240).

The root device will not have a port number and in such case it's ok to keep the port number at 0 and the path empty.

Reply:
Can't believe quite how I missed that. Good shout.